prymer/api/picking.py

```diff
@@ -140,13 +141,12 @@ def build_primer_pairs(
         # generate all the primer pairs that don't violate hard size and Tm constraints
         for lp in left_primers:
             for rp in right_primers:
-                if rp.span.end - lp.span.start + 1 > amplicon_sizes.max:
+                amp_span = PrimerPair.calculate_amplicon_span(lp, rp)
+
+                if amp_span.length > amplicon_sizes.max:
                     continue
 
-                amp_mapping = Span(refname=target.refname, start=lp.span.start, end=rp.span.end)
-                amp_bases = bases[
-                    amp_mapping.start - region_start : amp_mapping.end - region_start + 1
-                ]
+                amp_bases = bases[amp_span.start - region_start : amp_span.end - region_start + 1]
```

Comment:
**issue/question** `Span` is one-based inclusive, does the start position need to be offset by 1 to correctly index into the zero-based string?

Reply:
It works out because `region_start` is also 1-based (we subtract 1 from it when fetching from the fasta above).  So `amp_span.start - region.start` becomes a 0-based offset into the string.

Reply:
Could we leave a comment to that effect so it's clearer for future readers?

Reply:
Done.